src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,107 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
+import * as clientComments from '../service/comment';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type Props = {
+  createComment: (comment: Comment) => void
+  post: Post
+};
+
+export const NewCommentForm: React.FC<Props> = ({ createComment, post }) => {
+  const [nameText, setNameText] = useState('');
+  const [emailText, setEmailText] = useState('');
+  const [commentText, setCommentText] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [commentError, setCommentError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const reset = () => {
+    setNameText('');
+    setEmailText('');
+    setCommentText('');
+    setNameError(false);
+    setEmailError(false);
+    setCommentError(false);
+    setIsLoading(false);
+  };
+
+  const addComment = ({
+    postId, name, email, body,
+  }: Omit<Comment, 'id'>) => {
+    setIsLoading(true);
+
+    clientComments.createComment({
+      postId, name, email, body,
+    })
+      .then(createComment)
+      .finally(() => {
+        setCommentText('');
+        setIsLoading(false);
+      });
+  };
+
+  const changeNameText = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (nameText.length === 0) {
+      setNameError(true);
+    }
+
+    setNameError(false);
```

Comment:
is it works as expected? it's just immediately becomes false after true

Reply:
Yeah, it's working how I expected. If I write another, with return it doesn't working correctly(